sm_common/test/assert_macros_always_assert.cpp

```diff
@@ -0,0 +1,13 @@
+#include <gtest/gtest.h>
+
+#ifndef NDEBUG
+#define NDEBUG
+#endif
+#define SM_ALWAYS_ASSERT
```

Comment:
I guess this requires another `#ifndef` because later it might be enabled already. But that is not important.